Mathlib/Data/Rat/NNRat.lean

```diff
@@ -0,0 +1,491 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+
+! This file was ported from Lean 3 source module data.rat.nnrat
+! leanprover-community/mathlib commit 28aa996fc6fb4317f0083c4e6daf79878d81be33
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Algebra.Basic
+import Mathlib.Algebra.Order.Nonneg.Field
+
+/-!
+# Nonnegative rationals
+
+This file defines the nonnegative rationals as a subtype of `Rat` and provides its algebraic order
+structure.
+
+We also define an instance `CanLift ℚ ℚ≥0`. This instance can be used by the `lift` tactic to
+replace `x : ℚ` and `hx : 0 ≤ x` in the proof context with `x : ℚ≥0` while replacing all occurrences
+of `x` with `↑x`. This tactic also works for a function `f : α → ℚ` with a hypothesis
+`hf : ∀ x, 0 ≤ f x`.
+
+## Notation
+
+`ℚ≥0` is notation for `NNRat` in locale `NNRat`.
+-/
+
+
+open Function
+
+open BigOperators
+
+/-- Nonnegative rational numbers. -/
+def NNRat := { q : ℚ // 0 ≤ q } deriving
+  CanonicallyOrderedCommSemiring, CanonicallyLinearOrderedSemifield, LinearOrderedCommGroupWithZero,
+  Sub, Inhabited  -- Porting note: Removed `OrderedSub, DenselyOrdered, Archimedean`
```

Comment:
Why did you remove these? Are they implied by the others?

Reply:
They were causing errors. I used `instance` to try to add them back.